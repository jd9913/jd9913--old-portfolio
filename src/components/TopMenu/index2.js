import React from 'react';


function Nav() {

    const categories=[
        {
            name:"highlights",
            description: "The projects that I've created or are working on"
        },
        {
            name: "resume",
            description: "a listing of my work experience"
        },
        {
            name:"education",
            description: "things I've done to better myself"
        },
    ];

function categorySelected(){
    console.log("This is my profile page!")
}

    return (

      <header>  
          <h2>
          <a data-testid='link' href="/">
              <span role="img" aria-label="sparkles" src="../../assets/images/logo/sparkles_2728.png">âœ¨Jennifer!</span> 
          </a>
          </h2> 
          <nav>
              <ul>
                  <li className="mx-2">
                      <a data-testid='about' href="#about">
                          About Me
                      </a>
                  </li>
                  <li>
                      <span>Contact</span>
                  </li>
                  {categories.map((category)=>(
                      <li className="mx-1"
                      key={category.name}
                      >
                          <span onClick={categorySelected} >{category.name}</span>
                      </li>
                  ))}
              </ul>
          </nav> 
        </header>
    );


}

export default Nav;
