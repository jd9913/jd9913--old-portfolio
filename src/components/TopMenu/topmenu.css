

#branding{
    text-align: center;
    font-size: 38px;
    color: var(--cloudBurst);
    font-family:cursive;
    
}


.jumbotron{
    text-align: center;
    background: "../../assets/images/self/jennifer-Schlierman-logo3.png";
}

.navigation{
    width: 85%;
    max-width: 1200px;
    margin-left: auto;
    margin-right: auto;
    color: var(--satinLinen);
    
}