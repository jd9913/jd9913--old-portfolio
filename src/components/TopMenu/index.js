import React from "react";
import "./topmenu.css";

import Jumbotron from "react-bootstrap/Jumbotron";
import Button from "react-bootstrap/Button";

function TopMenu() {
  const categories = [
    {
      name: "Highlights",
      description: "The projects that I've created or are working on",
      link: "#highlights",
      test: "highlight",
    },
    {
      name: "Contact",
      description: "Reach out to me",
      link: "#contact-form",
      test: "contact",
    },
    {
      name: "About Me",
      description: "Who am I",
      link: "#about",
      test: "about",
    },
    // {
    //     name: "resume",
    //     description: "a listing of my work experience"
    // },
    // {
    //     name:"education",
    //     description: "things I've done to better myself"
    // },
  ];

  function categorySelected() {
    window.location = categories.link;
    // alert("you clicked a button!");
  }

  return (
    <section>
      <Jumbotron
        className="branding grid-container"
        data-testid="link"
        href="/"
      >
        <p>
          <span
            role="img"
            aria-label="sparkles"
            src="../../assets/images/logo/sparkles_2728.png"
          >
            <h1>âœ¨Jennifer!</h1>
          </span>
        </p>
      </Jumbotron>
      <div className="navigation">
        {categories.map((category, index) => (
          <Button href={category.link} key={index} data-testid={category.test} variant="primary" size="lg" id="button">
            <span>{category.name}</span>{" "}
          </Button>
        ))}
      </div>
    </section>
  );
}

export default TopMenu;
