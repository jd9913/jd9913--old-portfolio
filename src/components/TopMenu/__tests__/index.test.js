import React from 'react';
import {render, cleanup} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import TopMenu from '..';

afterEach(cleanup);

describe('Nav component', ()=>{
    it('renders', ()=>{
        render(<TopMenu/>);
    });

    it('matches snapshot', ()=>{
        const { asFragment } = render(<TopMenu/>);
        expect (asFragment()).toMatchSnapshot();
    })

  
})


describe('emoji is visible', ()=>{
    it('inserts emoji into the h2', ()=>{
        const { getByLabelText }=render(<TopMenu/>);
        expect(getByLabelText('sparkles')).toHaveTextContent('âœ¨')
    })

});

describe('links are visible', ()=>{
    it('inserts text into the links', ()=>{
        const { getByTestId }=render(<TopMenu/>);
        expect(getByTestId('link')).toHaveTextContent('Jennifer!');
        expect(getByTestId('about')).toHaveTextContent('About Me');


    });
})