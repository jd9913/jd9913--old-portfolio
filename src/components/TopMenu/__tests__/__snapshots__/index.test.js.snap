// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav component matches snapshot 1`] = `
<DocumentFragment>
  <section>
    <div
      class="branding grid-container jumbotron"
      data-testid="link"
      href="/"
    >
      <p>
        <span
          aria-label="sparkles"
          role="img"
          src="../../assets/images/logo/sparkles_2728.png"
        />
      </p>
      <h1>
        âœ¨Jennifer!
      </h1>
      <p />
    </div>
    <div>
      <a
        class="btn btn-primary btn-lg"
        data-testid="highlight"
        href="#highlights"
        id="button"
      >
        <span>
          Highlights
        </span>
         
      </a>
      <a
        class="btn btn-primary btn-lg"
        data-testid="contact"
        href="#contact-form"
        id="button"
      >
        <span>
          Contact
        </span>
         
      </a>
      <a
        class="btn btn-primary btn-lg"
        data-testid="about"
        href="#about"
        id="button"
      >
        <span>
          About Me
        </span>
         
      </a>
    </div>
  </section>
</DocumentFragment>
`;
